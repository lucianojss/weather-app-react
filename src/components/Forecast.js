import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import WeatherIcon from 'react-icons-weather';

const styles = {
    card: {
        maxWidth: '100%',
        marginBottom: '10px'
    },
    cardContent: {
        textAlign: 'center'
    },
    icon: {
		margin: 10,
		fontSize: '2em'
	}
};

class Forecast extends Component {
    render() {
        const { classes, code, date, high, low, text, units } = this.props;
        return (
            <Card className={classes.card}>
                <CardContent className={classes.cardContent}>
                    <Typography variant="subheading">{date}</Typography>
                    <Typography>Low {low} ยบ{units} | Max {high} ยบ{units}</Typography>
                    <WeatherIcon className={classes.icon} name="yahoo" iconId={code}/>
                    <Typography variant="subheading">{text}</Typography>
                </CardContent>
            </Card>
        );
    }
}

Forecast.propTypes = {
    classes: PropTypes.object.isRequired,
    code: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    day: PropTypes.string,
    high: PropTypes.string.isRequired,
    low: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    units: PropTypes.string.isRequired
};

export default withStyles(styles)(Forecast);
