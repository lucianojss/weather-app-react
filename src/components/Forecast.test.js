import React from 'react';
import Forecast from './Forecast';
import renderer from 'react-test-renderer';

describe('#Forecast', () => {
    test('Should render Forecast Component correctly', () => {

        const forecast = {
            code: '19',
            date: 'random Date',
            high: '26',
            low: '-50',
            text: 'random text',
            units: 'C'
        };

        const component = renderer.create(
            <Forecast {...forecast}></Forecast>,
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

    });
});
