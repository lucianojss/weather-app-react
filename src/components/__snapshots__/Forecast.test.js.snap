// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Forecast Should render Forecast Component correctly 1`] = `
<div
  className="MuiPaper-root-5 MuiPaper-elevation1-8 MuiPaper-rounded-6 MuiCard-root-4 Forecast-card-1"
>
  <div
    className="MuiCardContent-root-32 Forecast-cardContent-2"
  >
    <h3
      className="MuiTypography-root-33 MuiTypography-subheading-40"
    >
      random Date
    </h3>
    <p
      className="MuiTypography-root-33 MuiTypography-body1-42"
    >
      Low 
      -50
       ยบ
      C
       | Max 
      26
       ยบ
      C
    </p>
    <i
      className="wi wi-yahoo-19 Forecast-icon-3"
      name="yahoo"
    />
    <h3
      className="MuiTypography-root-33 MuiTypography-subheading-40"
    >
      random text
    </h3>
  </div>
</div>
`;
